src/main/java/com/pontusvision/graphutils/globals.groovy

```diff
@@ -2333,6 +2334,107 @@ class VisJSGraph {
 
 }
 
+class Utils {
+
+  // signature with Strings as params
+  static Long mergeVertices(String srcVid1, String targetVid2,  boolean removeRid1Vertex, boolean removeRid1Edge) {
+    return mergeVertices(new ORecordId(srcVid1), new ORecordId(targetVid2), removeRid1Vertex, removeRid1Edge);
+
+  }
+
+  // main method
+  static Long mergeVertices(ORID srcVid1, ORID targetVid2,  boolean removeRid1Vertex, boolean removeRid1Edge) {
+    //Inputs:
+    //  rid1(source); rid2(target); removeRid1Edge(boolean)
+    /*
+          A1            B1====B2
+          |\____A2
+          |
+          A3
+
+          A1, B1, remove=true
+
+          B1=====B2
+          |\______A2
+          |
+          A3
+
+   */
+
+    //Algorithm:
+    //  for each edge in bothrids (to/from) rid1:
+    //    get the edge label + the neighbour id to / from
+    //    if (to is the same as Rid1):
+    //      create a new edge from rid2 to the neighbour
+    //    else
+    //      create a new edge from neighbour to rid2
+    //    endif
+    //    if (removeRid1Edge):
+    //      delete edge
+    //    endif
+    //  endfor
+    //output:
+    //  numberOfEdges moved/copied
+
+    def tx = App.g.tx();
+
+    if (!tx.isOpen()) {
+      tx.open()
+    }
+
+    Long numberOfEdges = 0;
+
+    try {
+
+      def delEdges = [];
+
+      App.g.V(srcVid1).bothE().each { Edge e ->
+        Vertex fromInV = e.inVertex()
+        Vertex toOutV = e.outVertex()
+        String label = e.label()
+
+        numberOfEdges++;
+
+        ORID fromInVRid = fromInV.id();
+        ORID toOutVRid = toOutV.id();
+
+        if (toOutVRid.equals(srcVid1)) {
```

Comment:
we need to check whether there's already an edge with the same label between the two vertices before creating a new one.  Otherwise, we will end up with duplicate edges